src/SlidingSyncManager.ts

```diff
@@ -359,7 +359,7 @@ export class SlidingSyncManager {
         let proxyUrl: string | undefined;
 
         try {
-            const clientWellKnown = await AutoDiscovery.findClientConfig(client.getDomain()!);
+            const clientWellKnown = await AutoDiscovery.findClientConfig(client.getDomain());
```

Comment:
`AutoDiscovery.findClientConfig` needs a string, not null, so I think the right fix here would be to null-check `client.getDomain()` and throw an exception.